net/graphsync_fetcher_test.go

```diff
@@ -522,12 +392,108 @@ type requestResponse struct {
 }
 
 type mockableGraphsync struct {
+	ctx              context.Context
+	builder          *chain.Builder
```

Comment:
Consume an interface (extend `MessageStore`+`BlockProvider`) interface rather than this concrete type (in `tryBlockMessageReceiptNode` too)